### DISCUSSION TITLE
`In` replaced by `HasSome` and `HasMany` confusion for non-slice columns

### DISCUSSION THREAD

**Discussion by liamross** [prisma/prisma-client-go#644] [Q&A] [answered]
If I have a column, let's say `id`, and I want to `FindMany` of something based on a list of IDs I have, I previously used `In`. However, it seems that was removed in the new release.

```go
var ids []string // List of IDs to find

entries, err := database.SomeEntry.FindMany(
        db.SomeEntry.ID.In(ids),
).Exec(context.Background())
```

Here are my issues:

1. `In` made a lot of sense to me, but no longer exists (read as `FindMany` where ID is `In` a slice of IDs)
2. `HasEvery` is now an option... what does that mean? How can an ID have every value of a slice? Does this mean only `FindMany` if every single ID from the slice is found?
3. `HasSome` is presumably the new version of `In`? But that's far less intuitive.

Questions:

1. What is the new version of `In`?
2. Do `HasEvery` and `HasSome` really make sense here? If so could someone explain the right way to think about them, that would be very appreciated.

```go
entries, err := database.SomeEntry.FindMany(
        db.SomeEntry.ID.HasSome(ids), // Is this even the same? What does it all mean :'(
).Exec(context.Background())
```

**Comment by steebchen** [comment, marked as answer]
This seems like a bug to me. Can you please share your schema?

**Reply by liamross** [reply]
I can't share the schema but I'll try to do a minimal reproduction, I actually reverted back to `v0.11` to avoid this issue in the meantime so it's not currently a blocker. Will try to get reproduction asap

**Reply by liamross** [reply]
Reproduction sent in #643

**Comment by liamross** [comment]
This is a bug, see https://github.com/prisma/prisma-client-go/issues/643
